statsd/container_linux.go

```diff
@@ -153,6 +222,9 @@ func initContainerID(userProvidedID string, cgroupFallback bool) {
 			} else {
 				containerID = readMountinfo(selfMountInfoPath)
 			}
+			if containerID != "" {
```

Comment:
Note that we don't have a very reliable way to identify if we're inside a container. It means that even if the tracer is running on host, we'll send the inode of the cgroup

Reply:
Actually thinking more about that. If you are in cgroup host namespace, it's useless to get inode because `/proc/self/cgroup` is going yield a container id if present.

So I think it's fair to check for cgroup namesapce != host cgroup namespace to get `inode`. This will avoid some work for the lib and the Agent.
You can see how it's implemented from this code:
https://github.com/DataDog/datadog-agent/blob/main/pkg/util/containers/metrics/system/collector_linux.go#L84-L91

Reply:
It makes sense, I applied it in the last commit